tools/c7n_azure/c7n_azure/container_host/host.py

```diff
@@ -0,0 +1,379 @@
+
+        # Update event subscription
+        AzureEventSubscription.create(destination,
+                                      self.event_queue_name,
+                                      self.session.get_subscription_id(),
+                                      self.session, event_filter)
+
+        self.require_event_update = False
+
+    def poll_queue(self):
+        """
+        Poll the Azure queue and loop until
+        there are no visible messages remaining.
+        """
+        client = self.queue_service
+
+        # If we don't have any policies
+        # don't bother pulling events
+        if not self.policies:
+            return
+
+        while True:
+            messages = client.get_messages(
+                self.event_queue_name,
+                num_messages=queue_message_count,
+                visibility_timeout=queue_timeout_seconds)
+
+            if len(messages) == 0:
+                break
+
+            log.info('Pulled %s events to process while polling queue.' % len(messages))
+
+            for message in messages:
+                if message.dequeue_count > max_dequeue_count:
+                    Storage.delete_queue_message(client, self.event_queue_name, message=message)
+                    log.warning("Event deleted due to reaching maximum retry count.")
+                else:
+                    # Run matching policies
+                    self.run_policies_for_event(message)
+
+                    # We always dequeue the message.  We have
+                    # alternate options for retry if the behavior
+                    # is required
+                    Storage.delete_queue_message(
+                        client,
+                        self.event_queue_name,
+                        message=message)
+
+    def run_policies_for_event(self, message):
+        """
+        Find all policies subscribed to this event type
+        and schedule them for immediate execution.
+        """
+        # Load up the event
+        event = json.loads(base64.b64decode(message.content).decode('utf-8'))
+        operation_name = event['data']['operationName']
+
+        # Execute all policies matching the event type
+        for k, v in self.policies.items():
+            events = v['policy'].data.get('mode', {}).get('events')
+            if not events:
+                continue
+            events = AzureEvents.get_event_operations(events)
+            if operation_name in events:
+                self.scheduler.add_job(self.run_policy,
+                                       id=k + event['id'],
+                                       name=k,
+                                       args=[v['policy'],
+                                             event,
+                                             None],
+                                       misfire_grace_time=60 * 3)
+
+    def run_policy(self, policy, event, context):
+        try:
+            policy.validate()
```

Comment:
I think we should move validate piece to `update_policies` since we don't need to run it after we cached policy.